src/managers/RoleManager.js

```diff
@@ -128,9 +131,13 @@ class RoleManager extends CachedManager {
    *   .catch(console.error);
    */
   async create(options = {}) {
-    let { name, color, hoist, permissions, position, mentionable, reason } = options;
+    let { name, color, hoist, permissions, position, mentionable, reason, icon, unicodeEmoji } = options;
     if (color) color = resolveColor(color);
     if (typeof permissions !== 'undefined') permissions = new Permissions(permissions);
+    if (icon) {
+      const guildEmojiURL = this.guild.emojis.resolve(icon)?.url;
+      icon = guildEmojiURL ? await DataResolver.resolveImage(guildEmojiURL) : await DataResolver.resolveImage(icon);
```

Comment:
If the author shares a server with the client and uses an emoji from that server, this should be able to use that as well
```suggestion
      const emojiURL = this.client.emojis.resolve(icon)?.url;
      icon = emojiURL ? await DataResolver.resolveImage(emojiURL) : await DataResolver.resolveImage(icon);
```